client/packages/invoices/src/InboundShipment/api/hooks/line/useZeroInboundLinesQuantity.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  useTableStore,
+  useTranslation,
+  useDeleteConfirmation,
+} from '@openmsupply-client/common';
+import { useIsInboundDisabled } from '../utils/useIsInboundDisabled';
+import { useInboundRows } from './useInboundRows';
+import { useZeroInboundLineQuantity } from './useZeroInboundLineQuantity';
+
+export const useZeroInboundLinesQuantity = (): (() => void) => {
+  const { items, lines } = useInboundRows();
+  const { mutateAsync } = useZeroInboundLineQuantity();
+  const isDisabled = useIsInboundDisabled();
+  const t = useTranslation('replenishment');
+
+  const selectedRows =
+    useTableStore(state => {
+      const { isGrouped } = state;
+
+      return isGrouped
+        ? items
+            ?.filter(({ id }) => state.rowState[id]?.isSelected)
+            .map(({ lines }) => lines.flat())
+            .flat()
+        : lines?.filter(({ id }) => state.rowState[id]?.isSelected);
+    }) || [];
+
+  const notAllLinesSelected = selectedRows.length !== lines?.length;
+
+  const onZeroQuantities = async () => {
+    await mutateAsync(selectedRows).catch(err => {
+      throw err;
+    });
+  };
+
+  const confirmAndZeroLines = useDeleteConfirmation({
```

Comment:
Reusing the useDeleteConfirmation modal since it works for this... but should probably rename. Can't think of a name for it right now... will get back to it

Reply:
Just `useConfirmationDialog`? You would have to rename canDelete and I wouldn't fall back to the delete messages but make the message mandatory input parameters.

Maybe have another wrapper `useDeleteConfirmation` that calls `useConfirmationDialog` with the correct messages?

Reply:
Yeah, I was thinking of doing that but in a carry-over issue... touches quite a few place 🤔  Is that alright?

Reply:
If you keep a compatible version of `useDeleteConfirmation` that just wraps the new more general `useConfirmationDialog` hook you can keep the changes to a minimum.